packages/pools/src/router/routes.ts

```diff
@@ -161,18 +163,20 @@ export class OptimizedRoutes implements TokenOutGivenInRouter {
       throw new NoRouteError();
     }
 
-    // filter routes by enough entry liquidity
-    const routesInitialLimitAmounts = await Promise.all(
-      routes.map((route) =>
-        route.pools[0].getLimitAmountByTokenIn(tokenIn.denom)
-      )
-    );
-    routes = routes.filter((_, i) =>
-      routesInitialLimitAmounts[i].gte(tokenIn.amount)
-    );
+    // Only perform the filter if there are many routes.
+    // For a direct swap or, if there is only one route
+    // available, use it as is.
+    if (candidateRoutesLength > 1) {
```

Comment:
Remember: routes have multiple pools, should we also check that the one route has one pool?

Reply:
The core of the issue is that we might filter out the only possible route with a filter that is only useful for selecting between multiple routes. 

So I think it might be beneficial to keep it as is. WDYT?